### ISSUE TITLE
Expired certificate crashes logstash server

### ISSUE THREAD

**Issue by ranman** [logstash-plugins/logstash-input-tcp#35]
If you submit a message to logstash with an expired ssl certificate (either on the logstash server or on the client submitting the message):

```
SSL Error {:exception=>#<OpenSSL::SSL::SSLError: Received fatal alert: certificate_expired>, :backtrace=>["org/jruby/ext/openssl/SSLSocket.java:262:in `accept'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/jruby-openssl-0.9.12-java/lib/jopenssl19/openssl/ssl-internal.rb:106:in `accept'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-tcp-3.0.0/lib/logstash/inputs/tcp.rb:112:in `run_server'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-tcp-3.0.0/lib/logstash/inputs/tcp.rb:84:in `run'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.1.1-java/lib/logstash/pipeline.rb:206:in `inputworker'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.1.1-java/lib/logstash/pipeline.rb:199:in `start_input'"], :level=>:error, :file=>"logstash/inputs/tcp.rb", :line=>"117", :method=>"run_server"}
IOError: closed stream
                  peeraddr at org/jruby/ext/socket/RubyIPSocket.java:95
                  peeraddr at org/jruby/ext/openssl/SSL.java:209
                      peer at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.1.1-java/lib/logstash/util/socket_peer.rb:5
             handle_socket at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-tcp-3.0.0/lib/logstash/inputs/tcp.rb:171
  server_connection_thread at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-tcp-3.0.0/lib/logstash/inputs/tcp.rb:144
```

The logstash process will crash... should probably just reject the message.

**Comment by purbon** [comment]
@ranman can you open this at https://github.com/elastic/logstash indicating that this error make logstash completely to crash? This might also need some fixed upstream in logstash core pipeline as of better handling the exceptions coming over.

**Comment by jordansissel** [comment]
https://github.com/logstash-plugins/logstash-input-tcp/pull/43 should fix this, I think

**Comment by jordansissel** [comment]
#43 has been merged and released as plugin version 3.0.2. I'm going to close this issue since I think it's resolved.
